react/typings/events.d.ts

```diff
@@ -161,6 +162,14 @@ export interface PromotionClickData extends EventData {
   promotions: Promotion[]
 }
 
+export interface NewsletterSubscriptionData extends Exclude<EventData, 'currency'> {
```

Comment:
Why are you excluding the currency prop?

Reply:
I figured it wouldn't make sense in the context of the newsletter event, but I can keep the property.